src/js/components/Layer/Layer.js

```diff
@@ -1,78 +1,82 @@
-        this.originalFocusedElement.parentNode.focus
-      ) {
-        // required for IE11 and Edge
-        this.originalFocusedElement.parentNode.focus();
+const useLayerContainer = ({ animate, animation }) => {
+  const [layerContainer, setLayerContainer] = useState(null);
+
+  useEffect(() => {
+    const container = getNewContainer();
+    const originalFocusedElement = document.activeElement;
+
+    setLayerContainer(container);
+
+    return () => {
+      setLayerContainer(null);
+
+      if (originalFocusedElement) {
+        if (originalFocusedElement.focus) {
+          // wait for the fixed positioning to come back to normal
+          // see layer styling for reference
+          setTimeout(() => {
+            originalFocusedElement.focus();
+          }, 0);
+        } else if (
+          originalFocusedElement.parentNode &&
+          originalFocusedElement.parentNode.focus
+        ) {
+          // required for IE11 and Edge
+          originalFocusedElement.parentNode.focus();
+        }
       }
-    }
-
-    const activeAnimation = animation !== undefined ? animation : animate;
-    if (activeAnimation !== false) {
-      // undefined uses 'slide' as the default
-      // animate out and remove later
-      const layerClone = this.layerContainer.cloneNode(true);
-      layerClone.id = 'layerClone';
-      document.body.appendChild(layerClone);
-      const clonedContainer = layerClone.querySelector(
-        '[class*="StyledLayer__StyledContainer"]',
-      );
-      if (clonedContainer && clonedContainer.style) {
-        clonedContainer.style.animationDirection = 'reverse';
+
+      const activeAnimation = animation !== undefined ? animation : animate;
+      if (activeAnimation !== false) {
+        // undefined uses 'slide' as the default
+        // animate out and remove later
+        const layerClone = container.cloneNode(true);
+
+        layerClone.id = 'layerClone';
+        document.body.appendChild(layerClone);
+
+        const clonedContainer = layerClone.querySelector(
+          '[class*="StyledLayer__StyledContainer"]',
+        );
+        if (clonedContainer && clonedContainer.style) {
+          clonedContainer.style.animationDirection = 'reverse';
+        }
+        setTimeout(() => {
+          // we add the id and query here so the unit tests work
+          const clone = document.getElementById('layerClone');
+
+          if (clone) document.body.removeChild(clone);
+        }, animationDuration);
       }
-      setTimeout(() => {
-        // we add the id and query here so the unit tests work
-        const clone = document.getElementById('layerClone');
-        if (clone) document.body.removeChild(clone);
-      }, animationDuration);
-    }
-  }
-
-  render() {
-    const { islayerContainerAvailable } = this.state;
-
-    return islayerContainerAvailable
-      ? createPortal(<LayerContainer {...this.props} />, this.layerContainer)
-      : null;
-  }
-}
+    };
+  }, [animate, animation]);
```

Comment:
I think this means we will animate the Layer out and change the focus back if the user changes the `animate` or `animation` properties. While I don't think those are likely to change, something feels a bit odd about that behavior. It seems like we shouldn't touch the focus or animation unless the Layer is being unmounted. Would it be possible to leave the dependency array empty?

Reply:
Sure, I will do it.